### ISSUE TITLE
Setting a Git version 2.44 config pack.allowPackReuse to multi causes GitHub Desktop pushes to error

### ISSUE THREAD

**Issue by remlapmot** [desktop/desktop#18230]
### The problem

I think this is caused by a conflict between the Git I have installed (which is the latest Git version 2.44.0) and the version of Git within GitHub Desktop (which I believe is a bit older).

When I set the new feature in 2.44.0

```
git config --global pack.allowPackReuse multi
```

Pushes with GitHub Desktop fail with the message: 

> The remote disconnected. Check your internet connection and try again.

See screenshot below.

<img width="1275" alt="Screenshot 2024-02-28 at 15 04 34" src="https://github.com/desktop/desktop/assets/3777473/2c27ccd4-5b5f-4bac-b684-ca921d618c73">

Whereas pushing from the command line works.

When I set this back to the default

```
git config --global pack.allowPackReuse true
```

pushing works again as expected in GitHub Desktop.

(Setting `pack.allowPackReuse` to `single` also results in the same error message.)

I guess the simple answer is don't use features which are in a version of Git later than the version of Git within the GitHub Desktop app itself?

### Release version

3.3.9

### Operating system

macOS Sonoma 14.3.1

**Comment by steveward** [comment]
@remlapmot thanks for reporting this! I took a look at reproducing this just to confirm the underlying error:

```
2024-02-28T16:06:07.532Z - error: [ui] `git -c credential.helper= push origin main:main --progress` exited with an unexpected code: 1.
stderr:
fatal: bad boolean config value 'multi' for 'pack.allowpackreuse'
fatal: the remote end hung up unexpectedly
send-pack: unexpected disconnect while reading sideband packet
fatal: the remote end hung up unexpectedly
error: failed to push some refs to 'https://github.com/steveward/test-branch.git'

(The error was parsed as 4: The remote disconnected. Check your Internet connection and try again.)
```

I don't have a firm timeline for when we'll update to v2.44 from v2.39, but I'll bring this up for discussion in our next team sync.

**Comment by remlapmot** [comment]
Thanks very much

**Comment by sergiou87** [comment]
We can't fix the error (until we upgrade git to v2.44 or newer) but at least the error we show is more useful thanks to #18292

**Comment by remlapmot** [comment]
Thanks very much
